### PR TITLE
Refactors and updates sphinx documantation

### PR DESCRIPTION
**Pull request by StephenNneji**
- Updates packages for building docs  
- Makes an initial attempt to put the API documentation in order
- Updates installation information

### CODE REVIEW

File: docs/source/controlClass.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+2. **Differential Evolution** - Optimizes a problem by iteratively improving a candidate solution based on an evolutionary process
+3. **Nested Sampler** -  Estimates directly how the likelihood function relates to prior mass
+4. **DREAM**
+5. **Calculate** 
+
+Each of them has their own unique set of options. The fields in class object of controlsClass are 
+automatically updated based on the procedure selected.
+
+RAT uses parallelization to speed up the calculation. This is achieved using Parallel Computing Toolbox. The user can choose to use the parallelization or not and 
+if yes, what to parallelize on. (Points or Contrasts or all)
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply docs/source/controlClass.rst:20]: I am not sure "all" means both and now looking at the code "all" is never used

File: docs/source/backgroundsClass.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+5. `setBackground` - Set a background value in the background table. (Based on row)
+
+
+.. note::
+ 1. For constant, only one parameter is supplied to multi type table. 
+ 2. For data, only the data itself is supplied as a cell. 
+ 3. For function, the function name is supplied, along with up to three parameters (from the parameters table) which are then supplied to the function to
+    calculate the background. 
+    
+In each case, the background can either be added to the simulation or subtracted from the data.
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply docs/source/backgroundsClass.rst:47]: I think it means you want to add background to a sim but remove from experiment data

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply docs/source/backgroundsClass.rst:47]: Hmm, I think we need to clarify this either way. If they are different operations/affect different parts of the project this must be made clear.

File: docs/source/controlClass.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+to whether the users wants to calculate SLD during fit, and even how many iteration an algorithm should do etc.
+
+
+There are 5 different `controlsClass.procedures` that can be used with RAT. They are:
+
+1. **Simplex** 
+2. **Differential Evolution** - Optimizes a problem by iteratively improving a candidate solution based on an evolutionary process
+3. **Nested Sampler** -  Estimates directly how the likelihood function relates to prior mass
+4. **DREAM**
+5. **Calculate**
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply docs/source/controlClass.rst:14]: I will add the NS and DE in brackets in the docs for now we can make the refactor in another PR

**Review by DrPaulSharp — APPROVED** [review]

**Pull request merged**